src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_plugin.cpp

```diff
@@ -294,11 +297,12 @@ void AieTracePluginUnified::updateAIEDevice(void *handle, bool hw_context_flow)
       AIEData.metadata->getNumStreams(), AIEData.metadata->getHwContext(),
       AIEData.metadata);
 #elif XDP_VE2_BUILD
-  xrt::hw_context context = xrt_core::hw_context_int::create_hw_context_from_implementation(handle);
-  auto hwctx_hdl = static_cast<xrt_core::hwctx_handle*>(context);
-  auto hwctx_obj = dynamic_cast<shim_xdna_edge::xdna_hwctx*>(hwctx_hdl);
-  auto aieObj = hwctx_obj->get_aie_array();
-  XAie_DevInst* devInst = aieObj->get_dev();
+  XAie_DevInst* devInst = aie::trace::getAieDevInst(handle, hw_context_flow);
```

Comment:
Can we have a < TraceImpl >::"setAieDeviceInst" for which will call the "fetchAieDevInst" for its backend and populate VPStaticDb::aieDevInst ? For now, we can have it implemented for VE2 and Edge. This < TraceImpl >::"setAieDeviceInst" can be called before instantiating trace offloader, so that trace offloader can directly use the AieDevInst in StaticDB.